### PR TITLE
Move docker-init to /usr/libexec/docker/

### PR DESCRIPTION
**Pull request by cpuguy83**
Starting with docker v24 dockerd will lookup docker-init in more than just $PATH, including /usr/libexec/docker/*.
Moving it here is both more "correct" in terms of linux userspace layout, but also gets us away from needing to mark moby-engine < 24 as conflicting.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,12 +59,16 @@ $(COMPONENT_TARGETS):
 	$(DOCKER_BUILDX) bake $(PROGRESS) $@
 
 _LOCAL_PKG_DIR := $(OUTPUT)/$(DISTRO)
+ifneq ($(wildcard $(_LOCAL_PKG_DIR)/linux_*),)
+	_LOCAL_PKG_DIR := $(wildcard $(_LOCAL_PKG_DIR)/linux_*)
+endif
```
Comment (pmengelbert) [inline Makefile:64]: Why is this necessary?

File: Makefile [hunk 2]
```diff
@@ -59,12 +59,16 @@ $(COMPONENT_TARGETS):
...
 ifneq ($(wildcard $(_LOCAL_PKG_DIR)/*),)
 export LOCAL_PKG_DIR ?= $(_LOCAL_PKG_DIR)
 endif
 
 .PHONY: test
 test:
+	echo $(_LOCAL_PKG_DIR)
```
Comment (pmengelbert) [inline Makefile:71]: Left over?

**Review by pmengelbert — COMMENTED** [review]
Couple of questions

File: Makefile (hunk 2 above)
Comment (cpuguy83) [reply Makefile]: Fixed

File: Makefile (hunk 1 above)
Comment (cpuguy83) [reply Makefile:64]: Because the packages are under this dir.

Comment (cpuguy83) [reply Makefile:64]: Without this, the tests don't use the packages from bundles.

Comment (pmengelbert) [reply Makefile:64]: Hm, but the tests are working now without this change... the commits match during the containerd test for example. So the packages are making it into the aptly / local repo and getting installed. I'm fine with this change, just hope it doesn't break something.

**Review by pmengelbert — APPROVED** [review]
:+1:

File: Makefile (hunk 1 above)
Comment (cpuguy83) [reply Makefile:64]: Are you sure about that? The reason I noticed this is because I tried to force the `--init` case to fail by changing the init path to something nonsensical and the tests still passed.
Changing this allowed the test to fail.
It was also installing moby-engine 23 even though 24 was in bundles and as such not installing moby-tini.
